javascript/javascript1/week4/VoiceAssistant.js

```diff
@@ -0,0 +1,180 @@
+    { 
+        if (curUser.length) 
+        {
+            console.log(`Your name is: ${curUser}`);
+        } 
+        else 
+        {
+            console.log(`Sorry we don't recognize you`);
+        }
+    } 
+//Add todo list items
+    else if(command.toLowerCase().startsWith("add") &&
+    command.toLowerCase().endsWith("to my todo")) 
+    {
+
+        const task = command.replace("Add ", "").replace(" to my todo", "");
+        console.log("Added " + task + " to todo");
+        todos.push(task);
+    }
+//Remove todo list items 
+    else if(command.toLowerCase().startsWith("remove") &&
+    command.toLowerCase().endsWith("from my todo")) 
+    {
+        const task = command.replace("Remove ", "").replace(" from my todo", "");
+
+        let removed = false;
+        for(item in todos) 
+        {
+            if(todos[item] == task) 
+            {
+                todos.splice(item, 1);
+                console.log(`Removed ${task} to todo`);
+                removed = true;
+                break;
+            }
+        }
+
+        if (!removed) 
+        {
+            console.log(`Could not find ${task} in todo`);
+        }
+    }
+//Print todo list items 
+    else if(command.toLowerCase().includes("what is on my todo")) 
+    {
+        console.log("\nItems in Todo");
+        for(item in todos) 
+        {
+            console.log(" - " + todos[item]);
+        }
+    }
+//Simple arithmetic calculations
+    else if(command.toLowerCase().startsWith("what is"))
+    {
+        const calculateCommand = command.split(" ");
+        const firstNumber =  parseInt(calculateCommand[2]);
+        const secondNumber =  parseInt(calculateCommand[4]);
+        let result = 0;
+        if(calculateCommand[3] ===  '+')
+        {
+            result=firstNumber + firstNumber;
+            console.log(result);
+           
+        }
+       if(calculateCommand[3] === '-')
+        {
+            result=firstNumber - firstNumber;
+            console.log(result);
+        }
+        if(calculateCommand[3] === '*')
+        {
+            result=firstNumber * firstNumber;
+            console.log(result);
+        }
+        if(calculateCommand[3] === '/')
+        {
+            result=firstNumber / firstNumber;
+            console.log(result);
+        }
+        if(calculateCommand[3] === '%')
+        {
+            result=firstNumber % firstNumber;
+            console.log(result);
+        }
+        
+    }
+//Print current day in a human readable format
+    else if (command.includes("what day is today"))
+    {
+        const today = new Date().toDateString();
+        arrDate = today.split(" ");
+        console.log(`${arrDate[2]}. of ${arrDate[1]} ${arrDate[3]}`);
+    }
+//Set timer
+    else if(command.toLowerCase().startsWith("set timer for"))
+    {
+         
+        strCmd = command.split(" ");
+        timeUnit = strCmd.pop();
+        readTIme = strCmd[strCmd.length-1];
```

Comment:
These need to be assigned.

Reply:
I don't understand what "strCmd" means.

Reply:
You can use array destructuring here as well:
```js
const [ , , , readTime, timeUnit] = strCmd; // if you don't need the first three values, you can assign them to nothing like I did here

```

Reply:
Here different values to timeUnit and readTime, So is it possible to use array destructing here?

Reply:
The strCmd looks like this to begin with:
`["set", "timer", "for", "4", "minutes"]`

`strCmd.pop()` will return and remove the last element `"minutes"`.
Now the array looks like this:
`["set", "timer", "for", "4"]`

`strCmd[strCmd.length-1]` will return the (new) last element `"4"`.

When using array destructuring, you're directly assigning the elements to a variable:
```js
const [ setWord, timerWord, forWord, readTime, timeUnit] = strCmd;
// now "set" is assigned to setWord, "timer" to timerWord, etc.
```
 
Feel free to ask if you find it confusing :)

Reply:
Thank you so much for the great explanation.